tests/unit/rules/javascript/test.no_eval.js

```diff
@@ -52,7 +52,10 @@ describe('no_eval', () => {
     },
     {
       code: '(function(exe){ exe("foo") })(eval);',
-      message: ['eval can be harmful.'],
+      // The code snippet above will be parsed as script and we'll get the
+      // parsing as module error in the message. We use
+      // `expect.stringContaining()` to ignore the error message part.
+      message: [expect.stringContaining('eval can be harmful.')],
```

Comment:
(My apologies, I'm going to go back to discuss again about a point we previously discussed briefly in https://github.com/mozilla/addons-linter/pull/3556/files#r570419337).

I did want to double-check this one more time, and I think that we may actually have to fix this instead of making the test to pass:

- the test case related to this change is currently failing because the error produced by this tests becomes `"eval can be harmful. (Parsing as module error: Binding eval in strict mode at line: 1 and column: 14)"`, which I think it wrong because we are appending a parsing error to the message coming from another eslint rule that does not have anything to do with a parsing error (and also, given that the eslint rule have been able to produce this message, that means that at least we failed to parse it as an module but not as a script)

I think we may want to revent the changes applied to this change and tweak the changes applied to the `JavaScriptScanner` `scan` method to only add the parsing error to a fatal error without a ruleId (that should allow us to avoid to append the "parsing as a module" error to results coming from the eslint rules, and the parsing error doesn't include a ruleId), e.g. I think that something like this should do it and be reasonable enough:

```js
    results.forEach((result) => {
      result.messages.forEach((message) => {
        let extraShortDescription = [];
        // ...
        if (message.fatal === true && message.ruleId == null) {
          // ...
          if (detectedSourceType.parsingError) {
            const { type, error } = detectedSourceType.parsingError;
            extraShortDescription.push(`(Parsing as ${type} error: ${error})`);
          }

          // ...
          const formattedError = message.message.replace('Parsing error: ', '');
          extraShortDescription.push(
            oneLine`(Parsing as ${this.sourceType} error: ${formattedError} at
              line: ${message.line} and column: ${message.column})`
          );
          // eslint-disable-next-line no-param-reassign
          message.message = _messages.JS_SYNTAX_ERROR.code;
        }

        ...

        if (code === _messages.JS_SYNTAX_ERROR.code &&
            extraShortDescription.length) {
          shortDescription += ` ${extraShortDescription.join(' ')}`
        }
```

@willdurand what do you think? does that sounds like a reasonable approach?

Reply:
awww, I definitely overlooked that part. Yeah, that makes sense, sorry I should have caught that myself!

Reply:
@rpl hey, I see you added a check at the bottom of the snippet above:

```
if (code === _messages.JS_SYNTAX_ERROR.code &&
            extraShortDescription.length) {
```

Is it useful considering that we only update `extraShortDescription` where we set `code` to `_messages.JS_SYNTAX_ERROR.code`? I am thinking that we might want to reuse `extraShortDescription` in the future so maybe we don't need this new condition (and keep adding `extraShortDescription` when it has content)?

Reply:
(I updated this PR with the other changes)

Reply:
> @rpl hey, I see you added a check at the bottom of the snippet above:
> 
> ```
> if (code === _messages.JS_SYNTAX_ERROR.code &&
>             extraShortDescription.length) {
> ```
> 
> Is it useful considering that we only update `extraShortDescription` where we set `code` to `_messages.JS_SYNTAX_ERROR.code`? I am thinking that we might want to reuse `extraShortDescription` in the future so maybe we don't need this new condition (and keep adding `extraShortDescription` when it has content)?

@willdurand yeah, I was honestly on the fence on still checking the `extraShortDescription.length` there for similar reasons, I think it should be just fine if we do only check if extraShortDescription.length is not 0 instead.